src/noPipeImpureRule.ts

```diff
@@ -31,27 +32,18 @@ export class Rule extends AbstractRule {
 }
 
 export class ClassMetadataWalker extends NgWalker {
-  protected visitNgPipe(controller: ClassDeclaration, decorator: Decorator): void {
-    this.validatePipe(controller, decorator);
-    super.visitNgPipe(controller, decorator);
+  protected visitNgPipe(metadata: PipeMetadata): void {
+    this.validatePipe(metadata);
+    super.visitNgPipe(metadata);
   }
 
-  private validatePipe(controller: ClassDeclaration, decorator: Decorator): void {
-    const pureExpression = getDecoratorPropertyInitializer(decorator, 'pure');
+  private validatePipe(metadata: PipeMetadata): void {
+    if (!metadata.pure || metadata.pure!.kind !== SyntaxKind.FalseKeyword) return;
```

Comment:
```suggestion
    if (!metadata.pure || metadata.pure.kind !== SyntaxKind.FalseKeyword) return;
```

Reply:
You just missed the non null assertion 👀

Reply:
arff